graphbolt/src/csc_sampling_graph.cc

```diff
@@ -318,7 +318,7 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighborsImpl(
         auto num_picked_neighbors_data_ptr =
             num_picked_neighbors_per_node.data_ptr<scalar_t>();
         num_picked_neighbors_data_ptr[0] = 0;
-        const auto nodes_data_ptr = nodes.data_ptr<int64_t>();
+        const auto nodes_data_ptr = nodes.data_ptr<scalar_t>();
```

Comment:
Does it mean `seed nodes` should have same data type as `indices`?

Reply:
yes.

Reply:
But indptr holds offsets into edges, which should be a datatype that can hold the total number of edges. Indices holds vertex ids, which should be a datatype that can represent maximum vertex id. I would expect seed nodes to have the same type as the indices array.

Reply:
Then people can use int64 for `indptr` and int32 for `indices` if graph has less than 2B vertices but more than 2B edges.

Reply:
Now, I always format it to `int64_t` as it's for indexing only no matter the underlying data format is `int32_t` or `int64_t`. I think handling in this way is more general. What do you think?

Reply:
I think it is good, but we can do it via using a dispatch and use the original type of `nids` tensor, such as `const int64_t nid = nids[i]`;